docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+
+```swift
+extension Worker {
+  func work() -> EventLoopFuture<Value> { ... }
+  
+  #if swift(>=5.5)
+  func work() async throws -> Value { ... }
+  #endif
+}
+```
+
+If a function cannot fail but was using futures before, it should not include the throws keyword in its new incarnation. 
+
+Such adoption can begin immediately, and should not cause any issues to existing users of existing libraries. 
+
+### SwiftNIO helper functions
+
+To allow an easy transition to async code, SwiftNIO offers a number of helper methods on EventLoopFuture and -Promise. Those live in the `_NIOConcurrency` module and will move to NIOCore once Swift concurrency is released.
+
+On every `EventLoopFuture` you can call `.get()` to transition the future into an awaitable invocation. If you want to translate async/await calls to an EventLoopFuture we recommend the following pattern:
+
+```swift 
+#if swift(>=5.5)
+func yourAsyncFunctionConvertedToAFuture(on eventLoop: EventLoop) 
+    -> EventLoopFuture<Result> {
+    let promise = context.eventLoop.makePromise(of: Out.self)
+    promise.completeWithTask {
+        try await yourMethod(yourInputs)
+    }
+    return promise.futureResult
+}
+#endif
+```
+
+Further helpers exist for `EventLoopGroup`, `Channel`, `ChannelOutboundInvoker` and `ChannelPipeline`.
+
+### Sendable Checking
+
+> [SE-0302][SE-0302] introduced the Sendable protocol, which is used to indicate which types have values that can safely be copied across actors or, more generally, into any context where a copy of the value might be used concurrently with the original. Applied uniformly to all Swift code, Sendable checking eliminates a large class of data races caused by shared mutable state.
+>
+> -- from [Staging in Sendable checking][sendable-staging], which outlines the Sendable adoption plan for Swift 6.
+
+In the future we will see fully checked Swift concurrency. The language features to support this are the `Sendable` protocol and the `@Sendable` keyword for closures. Since sendable checking will break existing Swift code, a new major Swift version is required.
+
+To ease the transition to fully checked Swift code, it is possible to annotate your APIs with the Sendable protocol today.
+
+You can start adopting Sendable and getting appropriate warnings in Swift 5.5 already by passing the `-warn-concurrency` flag, you can do so in SwiftPM for the entire project like so:
+
+```
+swift build -Xswiftc -Xfrontend -Xswiftc -warn-concurrency
+```
+
+
+#### Sendable checking today
+
+Sendable checking is currently disabled in Swift 5.5(.0) because it was causing a number of tricky situations for which we lacked the tools to resolve.
+
+Most of these issues have been resolved on today’s `main` branch of the compiler, and are expected to land in the next Swift 5.5 releases. It may be worthwhile waiting with adoption until the next version(s) after 5.5.0.
```

Comment:
```suggestion
Most of these issues have been resolved on today’s `main` branch of the compiler, and are expected to land in the next Swift 5.5 releases. It may be worthwhile waiting for adoption until the next version(s) after 5.5.0.
```